### PR TITLE
HYB-824: Replace pageTimeout and noInternetConnection listeners for n…

### PR DESCRIPTION
**Pull request by kerrmarin**
Replaces the `pageTimeout` and `noInternetConnection` listener with a `navigationFailed` listener.

JIRA: https://mobify.atlassian.net/browse/HYB-824
Linked PRs: https://github.com/mobify/astro/pull/597

## Changes
- Replace the `pageTimeout` and `noInternetConnection` with the `naviagationFailed` event.

## How to test-drive this PR
- Point the scaffold to `HYB-824-remove-pageTimeout` branch of astro
- Run the app

and

1. Switch off internet and navigate to a new page
2. Use the network link conditioner to make your connection very bad and then navigate to a new page

### CODE REVIEW

File: app/app-controllers/error-screen/errorController.js [hunk 1]
```diff
@@ -89,7 +89,13 @@ define([
             self.viewPlugin.on('error:loaded', function() {
                 self.show();
             });
-            self.errorType = errorType;
+
+            if (eventArgs.error.code == -1001) { //timeout
```
Comment (jeremywiebe) [inline app/app-controllers/error-screen/errorController.js]: Once the Astro PR is merged we should change this to use the new js "constants" :)

Comment (kerrmarin) [inline app/app-controllers/error-screen/errorController.js]: Done 👍

File: app/app-controllers/error-screen/errorController.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ define([
         this.viewPlugin.off('back');
     };
 
-    ErrorController.prototype._generateErrorCallback = function(errorType, params) {
```
Comment (jeremywiebe) [inline app/app-controllers/error-screen/errorController.js]: Removing `errroType` here is breaking line [99](https://github.com/mobify/astro-scaffold/pull/125/files#diff-9e8c7e3e8735d4c25296f68a27665b19L99) below.

**Comment by jeremywiebe** [comment]
👍

**Pull request merged**